platforms/web/lib/dom.ts

```diff
@@ -538,6 +538,13 @@ function isNodeRequiringExtraOffset(node: Node) {
     // note this isn't simply a block node, we need to ensure that for lists we
     // do not add an extra offset for the ol/ul tag, only for the list items
     // themselves
+
+    // with the paragraph refactor, we could end up in a situation where we have
+    // a p tag containing formatted text next to a list, so if this is the case,
+    // we do want to add an extra offset _if_ the list has any children
+    if (node.nodeName === 'UL' || node.nodeName === 'OL') {
+        return node.hasChildNodes();
+    }
     return EXTRA_OFFSET_NODE_NAMES.includes(node.nodeName || '');
```

Comment:
Could `EXTRA_OFFSET_NODE_NAMES` be a bit misleading now as it doesn't contain `UL` or `OL`?

Reply:
I think that's a fair point. I'll have a look at the comments and naming in this section and see if they can be improved.

Reply:
Thanks for the comment, rereading this I've actually realised I could remove this special case and add the list nodes to the `EXTRA_OFFSET_NODE_NAMES` array. Much clearerf.